### DISCUSSION TITLE
How to retain MQTT entities available after HA restart ?

### DISCUSSION THREAD

**Discussion by Briochorama** [tesla-local-control/tesla_ble_mqtt_docker#6] [Q&A] [answered]
> [!NOTE]
> This is not an issue linked directly to the app, I think, but nevertheless I think it'd be useful to discuss it.

My problem is that, when HA restart, all MQTT entities provided by the app are marked `unavailable`. I've read a bit about MQTT (zero prior knowledge) and it [looks like it's related to the discovery/retention of topics](https://community.home-assistant.io/t/my-hand-built-mqtt-devices-are-unavailable-after-hass-io-restart-but-zigbee2mqtt-ones-are/201214).

Is this something we must configure in HA once the entities have been added ? For now, I just restart the container and they go back online.

**Comment by iainbullock** [comment]
Yes I noticed that. I will work on that for the next release. If you have a look at the dev branch I'm introducing a presence sensor. Some configuration of MQTT is required to achieve this which I will also use to address this problem

**Comment by iainbullock** [comment]
Reading https://www.home-assistant.io/integrations/mqtt/#mqtt-discovery I tihnk this can be achieved by either setting the retain flag or having my code respond to the LWT messages. The former is lazy and will result in 'ghost' entities if changes are made to container setup. I will look to implementing a response to the LWT message.

**Comment by iainbullock** [comment, marked as answer]
I have made another mod to the main branch without being able to test it properly. Bad practice I know, but it was another easy one.

I hope I've addressed the issue in this discussion. Release v1.0.10

Feedback welcome thanks

**Reply by Briochorama** [reply]
Rebooted HA twice, entities where available after each reboot.
